### PR TITLE
Improving minimization by grouping results based on arns

### PR DESCRIPTION
**Pull request by kvsaikiran**
## What does this PR do?
#235 
This will will further minimise the policy output by grouping the similar arns.

[//]: # (Required: Describe the effects of your pull request in detail. If
multiple changes are involved, a bulleted list is often useful.)

## What gif best describes this PR or how it makes you feel?
image

[//]: # (Encouraged: Insert an appropriate gif/meme to brighten up your reviewer's day.)

## Completion checklist

- [x] Additions and changes have unit tests
- [x] [Unit tests, Pylint, security testing, and Integration tests are passing.](https://github.com/salesforce/policy_sentry/actions). GitHub actions does this automatically
- [x] The pull request has been appropriately labeled using the provided PR labels

### CODE REVIEW

**Comment by kvsaikiran** [comment]
@kmcquade can you review this PR 😄

File: test/writing/test_minimize_write.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        write_format = sid_grp.process_template(cfg,minimize=1)
+        write_format.pop('Version')
+        self.assertEqual(len(write_format['Statement']), 2, "More than one statement returned, expected 1")
+        self.assertEqual(write_format['Statement'][0]['Action'], ['ssm:getpar*'], "extra actions are returned")
+        self.assertEqual(write_format['Statement'][0]['Resource'], cfg['read'], "Wrong resources were returned")
+        self.assertEqual(write_format['Statement'][1]['Action'], ['ssm:deletepar*', 'ssm:la*', 'ssm:putp*'],
+                         "extra actions are returned")
+        self.assertEqual(write_format['Statement'][1]['Resource'], cfg['write'], "Wrong resources were returned")
+
+if __name__ == '__main__':
```
Comment (reetasingh) [inline test/writing/test_minimize_write.py:50]: @kvsaikiran I dont think this main is required

File: test/writing/test_minimize_write.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import unittest
+from policy_sentry.writing.sid_group import SidGroup
+
+
```
Comment (reetasingh) [inline test/writing/test_minimize_write.py:4]: why a new file
there is already a test file for minimize https://github.com/salesforce/policy_sentry/blob/master/policy_sentry/writing/minimize.py

**Comment by kmcquade** [comment]
I love this GIF

File: test/writing/test_minimize_write.py (hunk 2 above)
Comment (kmcquade) [reply test/writing/test_minimize_write.py:4]: +1. It should be in the existing test file

**Comment by kvsaikiran** [comment]
@kmcquade @reetasingh please review again.

**Pull request merged**